src/clients/sig_rpc_tls_client.h

```diff
@@ -61,46 +25,17 @@ class SigHttpRpcTlsClient : public HttpRpcTlsClient
     const std::string& method, const nlohmann::json& params) override
   {
     const auto body_j = json_rpc(method, params);
-    const auto body_v = nlohmann::json::to_msgpack(body_j);
-    auto r = enclave::http::Request(HTTP_POST);
-    r.set_path(body_j["method"]);
-
-    tls::HashBytes body_digest;
-    tls::do_hash(body_v.data(), body_v.size(), body_digest, MBEDTLS_MD_SHA256);
-    const auto digest_value = fmt::format(
-      "{}={}",
-      "SHA-256",
-      tls::b64_from_raw(body_digest.data(), body_digest.size()));
-    r.set_header("digest", digest_value);
-
-    auto headers = r.get_headers();
-    headers["content-length"] = fmt::format("{}", body_v.size());
-
-    std::vector<std::string_view> headers_to_sign;
-    for (const auto& [k, v] : headers)
-    {
-      headers_to_sign.emplace_back(k);
-    }
+    const auto body_v = jsonrpc::pack(body_j, jsonrpc::Pack::Text);
 
-    std::string query = "";
-    const auto signing_string = enclave::http::construct_raw_signed_string(
-      http_method_str(HTTP_POST), method, query, headers, headers_to_sign);
-    if (!signing_string.has_value())
-    {
-      throw std::logic_error(fmt::format("Error constructing signed string"));
-    }
+    auto r = http::Request(body_j[jsonrpc::METHOD].get<std::string>());
+    r.set_header(
+      http::headers::CONTENT_TYPE, http::headervalues::contenttype::JSON);
 
-    const auto sig_raw = key_pair->sign(signing_string.value());
```

Comment:
This signing is now done inside `http::sign_request`, but this was the cause of #787 - this sign was _not_ using `sha256` to produce the digest. Now it is.